src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/ReferenceSearchParam.sql

```diff
@@ -1,13 +1,17 @@
-﻿CREATE TABLE dbo.ReferenceSearchParam
+﻿
+CREATE TABLE dbo.ReferenceSearchParam
 (
     ResourceTypeId                      smallint                NOT NULL,
     ResourceSurrogateId                 bigint                  NOT NULL,
     SearchParamId                       smallint                NOT NULL,
-    BaseUri                             varchar(128)            COLLATE Latin1_General_100_CS_AS NULL,
-    ReferenceResourceTypeId             smallint                NULL,
+    BaseUri                             varchar(128)            COLLATE Latin1_General_100_CS_AS NOT NULL CONSTRAINT DF_ReferenceSearchParam_BaseUri DEFAULT '',
+    ReferenceResourceTypeId             smallint                NOT NULL CONSTRAINT DF_ReferenceSearchParam_ReferenceResourceTypeId DEFAULT 0,
     ReferenceResourceId                 varchar(64)             COLLATE Latin1_General_100_CS_AS NOT NULL,
     ReferenceResourceVersion            int                     NULL,
     IsHistory                           bit                     NOT NULL,
+    CONSTRAINT PK_ReferenceSearchParam PRIMARY KEY NONCLUSTERED (ResourceTypeId, ResourceSurrogateId, SearchParamId, BaseUri, ReferenceResourceTypeId, ReferenceResourceId )
```

Comment:
>ResourceTypeId, ResourceSurrogateId, SearchParamId, BaseUri, ReferenceResourceTypeId, ReferenceResourceId [](http://example.com/codeflow?start=65&length=105)

I'm not expert, so I'll just summon one:
@SergeyGaluzo

Should we care about order of columns for primary key?
Is there are any difference between
`(int, int varchar(128), int, int,int)`
vs
`(int, int, int,int,int,varchar(128))`?

Reply:
We should not be adding new indexes, but rather modify old ones to convert them in PK. In this case I would add missing components to IXC_ReferenceSearchParam index, convert to clustered PK and remove current index.